cmd/run.go

```diff
@@ -34,7 +34,16 @@ log and return exit code of the command.`,
 		}
 
 		if err := e.Run(tableName, keyName, shellCommand, sleepStartRandom, holdLockBy); err != nil {
-			logrus.Fatal(err)
+			// dynamolock.LockNotGrantedError doesn't export the error so this is the next best we can do.
+			// This is obviously super fragile (i.e if the error msg wording is updated, this fails)
+			lockFailedErr := "Didn't acquire lock because it is locked and request is configured not to retry."
+
+			// Don't consider errors due to lock contention as failures
+			if err.Error() == lockFailedErr {
+				logrus.Warning(err)
+			} else {
+				logrus.Fatal(err)
+			}
```

Comment:
That's asking for problems, what you're looking for is:
```go
if !errors.Is(err, &dynamolock.LockNotGrantedError{}) {
  logrus.Fatal(err) /* this calls exit(1) */
}
logrus.Warning(err)

Reply:
Moved the logic per Shayon's req.